packages/email-pcd/src/EmailPCD.ts

```diff
@@ -74,7 +74,7 @@ export async function prove(args: EmailPCDArgs): Promise<EmailPCD> {
       argumentType: ArgumentTypeName.String
     },
     id: {
-      value: undefined,
+      value: args.id.value ? args.id.value + "-signature" : undefined,
```

Comment:
This is the primary fix here.  It took a lot more time to find than it took to implement. :)